File: common/src/main/java/mconi/common/sim/OniSimulationSnapshot.java
package mconi.common.sim;

public class OniSimulationSnapshot
{
	private final boolean running;
	private final long serverTicks;
	private final long simulationTicks;
	private final long lastSimulationTick;
	private final int tickInterval;
	private final int cellSize;
	private final int activeCells;

	public OniSimulationSnapshot(
			boolean running,
			long serverTicks,
			long simulationTicks,
			long lastSimulationTick,
			int tickInterval,
			int cellSize,
			int activeCells)
	{
		this.running = running;
		this.serverTicks = serverTicks;
		this.simulationTicks = simulationTicks;
		this.lastSimulationTick = lastSimulationTick;
		this.tickInterval = tickInterval;
		this.cellSize = cellSize;
		this.activeCells = activeCells;
	}

	public boolean running() { return running; }
	public long serverTicks() { return serverTicks; }
	public long simulationTicks() { return simulationTicks; }
	public long lastSimulationTick() { return lastSimulationTick; }
	public int tickInterval() { return tickInterval; }
	public int cellSize() { return cellSize; }
	public int activeCells() { return activeCells; }
}

File: common/src/main/kotlin/mconi/common/sim/OniSimulationSnapshot.kt
package mconi.common.sim

class OniSimulationSnapshot(
    private val running: Boolean,
    private val serverTicks: Long,
    private val simulationTicks: Long,
    private val lastSimulationTick: Long,
    private val tickInterval: Int,
    private val cellSize: Int,
    private val activeCells: Int,
    private val powerGenerationW: Double,
    private val powerDemandW: Double,
    private val storedEnergyJ: Double,
    private val powerTripped: Boolean,
    private val colonyStress: Double,
) {
    fun running(): Boolean = running
    fun serverTicks(): Long = serverTicks
    fun simulationTicks(): Long = simulationTicks
    fun lastSimulationTick(): Long = lastSimulationTick
    fun tickInterval(): Int = tickInterval
    fun cellSize(): Int = cellSize
    fun activeCells(): Int = activeCells
    fun powerGenerationW(): Double = powerGenerationW
    fun powerDemandW(): Double = powerDemandW
    fun storedEnergyJ(): Double = storedEnergyJ
    fun powerTripped(): Boolean = powerTripped
    fun colonyStress(): Double = colonyStress
}
